docs/flutterweathertutorial.md

```diff
@@ -43,476 +67,2514 @@ We'll be focusing on two endpoints:
+
+Create a new directory on the project level called `packages`. This directory will store all of our custom made subpackages.
+
+Within this directory, run the built in flutter create command to create a new subpackage called `meta_weather_api` for our api client
+
+```bash
+flutter create --template=package meta_weather_api 
+```
+
+
+```
+flutter_weather
+|-- lib/
+|-- test/
+|-- packages/
+  |-- meta_weather_api/
+    |-- lib/
+    |-- test/
+```
+
+### Creating Our MetaWeather API Data Model
+
+> Next, let's create `location.dart` and `weather.dart` models to store response data from the location and weather API endpoints.
+
+```
+flutter_weather
+|-- lib/
+|-- test/
+|-- packages/
+  |-- meta_weather_api/
+    |-- lib/
+      |-- src/
+        |-- models/
+          |-- location.dart
+          |-- weather.dart
+    |-- test/
+```
+
+> First, let's work out of `location.dart`. Our location model should store data returned by the location API, which looks like the following: 
+
+```json
+{
+    "title": "London",
+    "location_type": "City",
+    "woeid": 44418,
+    "latt_long": "51.506321,-0.12714"
+}
+```
+
+> Here's the **in-progress** `location.dart` file which stores the above response:
+
+```dart
+// packages/meta_weather_api/lib/src/models/location.dart (in progress)
+
+enum LocationType {
+    city, region, state, province, country, continent
+}
+
+class Location {
+    const Location({
+        required this.title,
+        required this.locationType,
+        required this.latLng,
+        required this.woeid,
+    });
+
+    final String title;
+    final LocationType locationType;
+    final LatLng latLng;
+    final int woeid;
+}
+
+class LatLng {
+    const LatLng({required this.latitude, required this.longitude});
+
+    final double latitude;
+    final double longitude;
+}
+
+```
+
+> Next, let's work out of `weather.dart`. Our weather model should store data returned by the weather API, which looks like the following:
+
+```json
+{
+    "id": 5037922198749184,
+    "weather_state_name": "Heavy Cloud",
+    "weather_state_abbr": "hc",
+    "wind_direction_compass": "SSE",
+    "created": "2021-05-28T15:32:01.902125Z",
+    "applicable_date": "2021-05-28",
+    "min_temp": 9.61,
+    "max_temp": 19.375,
+    "the_temp": 18.54,
+    "wind_speed": 3.0192401717198227,
+    "wind_direction": 148.63313166521016,
+    "air_pressure": 1023.0,
+    "humidity": 57,
+    "visibility": 10.56983466555317,
+    "predictability": 71
+}
+```
+
+> Here's the **in-progress** `weather.dart` file which stores the above response:
+
+```dart
+// packages/meta_weather_api/lib/src/models/weather.dart (in progress)
+
+enum WeatherState {
+    snow, sleet, hail, thunderstorm, heavyRain, lightRain, showers, heavyCloud, lightCloud, clear, unknown
+}
+
+enum WindDirectionCompass {
+    north, northEast, east, southEast, south, southWest, west, northWest, unknown
+}
```

Comment:
can we format these to 80 characters so that developers don't have to scroll horizontally to see all values?